modules/programs/zsh.nix

```diff
@@ -190,17 +202,22 @@ in
       envVarsStr = concatStringsSep "\n" (
         mapAttrsToList export config.home.sessionVariables
       );
+
+      zdotdir = "$HOME/" + cfg.dotDir;
     in mkIf cfg.enable {
       home.packages = with pkgs; [ zsh ]
         ++ optional cfg.enableCompletion nix-zsh-completions
         ++ optional cfg.oh-my-zsh.enable oh-my-zsh;
 
+      home.sessionVariables.ZDOTDIR = if cfg.dotDir != "" then zdotdir else null;
```

Comment:
My zsh breaks when I test this module with no `cfg.dotDir` specified. It seems like it happens because ZDOTDIR is set to empty string with the default configuration. Looks like we should set this variable only if the user specified something in `cfg.dotDir`.